src/Uno.Toolkit.UI/Controls/ZoomContentControl/ZoomContentControl.cs

```diff
@@ -0,0 +1,782 @@
+		set => SetValue(HorizontalMaxScrollProperty, value);
+	}
+
+	public double HorizontalMinScroll
+	{
+		get => (double)GetValue(HorizontalMinScrollProperty);
+		set => SetValue(HorizontalMinScrollProperty, value);
+	}
+
+	public double HorizontalScrollValue
+	{
+		get => (double)GetValue(HorizontalScrollValueProperty);
+		set => SetValue(HorizontalScrollValueProperty, value);
+	}
+
+	public double VerticalScrollValue
+	{
+		get => (double)GetValue(VerticalScrollValueProperty);
+		set => SetValue(VerticalScrollValueProperty, value);
+	}
+
+	public double ViewPortHeight
+	{
+		get => (double)GetValue(ViewPortHeightProperty);
+		set => SetValue(ViewPortHeightProperty, value);
+	}
+
+	public double ViewPortWidth
+	{
+		get => (double)GetValue(ViewPortWidthProperty);
+		set => SetValue(ViewPortWidthProperty, value);
+	}
+
+	#endregion
+
+	private async Task RaiseRenderedContentUpdated()
+	{
+		await Task.Yield();
+		RenderedContentUpdated?.Invoke(this, EventArgs.Empty);
+	}
+
+	private void RegisterPropertyHandlers()
+	{
+		// Register for property changed events.
+		RegisterPropertyChangedCallback(ZoomLevelProperty, OnZoomLevelChanged);
+		RegisterPropertyChangedCallback(MinZoomLevelProperty, CoerceZoomLevel);
+		RegisterPropertyChangedCallback(MaxZoomLevelProperty, CoerceZoomLevel);
+
+		RegisterPropertyChangedCallback(HorizontalOffsetProperty, OnHorizontalOffsetChanged);
+		RegisterPropertyChangedCallback(VerticalOffsetProperty, OnVerticalOffsetChanged);
+
+		RegisterPropertyChangedCallback(IsActiveProperty, IsActiveChanged);
+
+		this.SizeChanged += (s, e) => UpdateContentBoundsVisibility();
+	}
+
+	private void OnVerticalOffsetChanged(DependencyObject sender, DependencyProperty dp)
+	{
+		UpdateContentBoundsVisibility();
+		UpdateScrollVisibility();
+		UpdateVerticalScrollBarValue();
+	}
+
+	private void OnHorizontalOffsetChanged(DependencyObject sender, DependencyProperty dp)
+	{
+		UpdateContentBoundsVisibility();
+		UpdateScrollVisibility();
+		UpdateHorizontalScrollBarValue();
+	}
+
+	private async void OnZoomLevelChanged(DependencyObject sender, DependencyProperty dp)
+	{
+		CoerceZoomLevel(sender, dp);
+		UpdateScrollLimits();
+		UpdateContentBoundsVisibility();
+		UpdateScrollVisibility();
+		await RaiseRenderedContentUpdated();
+	}
+
+	public record BoundsVisibilityData
+	{
+		public static readonly BoundsVisibilityData None = new BoundsVisibilityData(leftEdge: false, topEdge: false, rightEdge: false, bottomEdge: false);
+
+		public bool LeftEdge { get; init; }
+		public bool TopEdge { get; init; }
+		public bool RightEdge { get; init; }
+		public bool BottomEdge { get; init; }
+		public bool AllEdgesVisible { get; init; }
+
+		public BoundsVisibilityData(bool leftEdge, bool topEdge, bool rightEdge, bool bottomEdge)
+		{
+			LeftEdge = leftEdge;
+			TopEdge = topEdge;
+			RightEdge = rightEdge;
+			BottomEdge = bottomEdge;
+			AllEdgesVisible = LeftEdge && TopEdge && RightEdge && BottomEdge;
+		}
+	}
```

Comment:
```suggestion
	public record BoundsVisibilityData(bool LeftEdge, bool TopEdge, bool RightEdge, bool BottomEdge)
	{
		public static BoundsVisibilityData None => new BoundsVisibilityData(false, false, false, false);
		
		public bool AllEdgesVisible => LeftEdge && TopEdge && RightEdge &&BottomEdge;
	}
```

Reply:
almost move this class to the bottom of this file, you shouldnt declare type in the middle of methods